### ISSUE TITLE
Understanding "Desired error not necessarily achieved due to precision loss"

### ISSUE THREAD

**Issue by libalk** [tBuLi/symfit#141]
I just found symfit and it seems to do exactly what I need! 

Just to test it out I started with the "Simple example" in the tutorial, and I'm getting the "Fitting status message: Desired error not necessarily achieved due to precision loss". I also noticed that the parameters, standard deviation, number of iterations and regression coefficient change every time I run.

So, what does this fitting status mean and how should I solve this?

(in case it matters, I am using jupyter notebook on windows)

**Comment by libalk** [comment]
Sorry, I should really check my code before posting an issue...
Still don't know what the status means, but the changing results was my own fault for rerunning the whole example (and not just the fit...)

**Comment by pckroon** [comment]
Hey Libalk,

I'm happy to see you managed to solve your issue. I'll try to provide a bit more insight on what I *think* is happening. 
There's two things going on: firstly, fitting is a minimization (usually we minimize a chi squared). Many minimizers will call this the error. A minimizer will also have a certain goal in mind, how small does the error have to get before we consider it small enough? (IIRC you can control this by passing arguments to `execute`). 
The second thing has to do with how a computer stores numbers. It simply can't store numbers with infinite precision: floating point numbers will get truncated at some point (and because of this floating point operations make errors). So as more and more floating point operations are performed, precision is lost.

To sum up: the minimizer has accumulated so many tiny mistakes in how it deals with numbers it's no longer sure whether it's succesfully finished.

If you want more feedback please post your exact code.

Peter

**Comment by tBuLi** [comment]
I agree with everything @pckroon says. 

But I am now curious if MINPACK suffers from the same problem, because this status message is quite common with minimizers such as BFGS, but I don't recall seeing this as often when MINPACK was still the default minimizer.

Perhaps you could update your code to something like

```python
from symfit.core.minimizers import MINPACK

Fit(model, x=xdata, y=ydata, minimizer=MINPACK)
```
where I ignored all other code necessary to make this work.

I would be curious to see if this changes the status message.
